File: Difficulty: Easy/Square Root/square-root.java
class Solution {
    int floorSqrt(int n) {
        // code here
        long low = 1;
        long high = n;
        
        while(low<=high) {
            long mid = low + (high - low) / 2;
            long answer = mid * mid;
            
            if(answer <= n) {
                low = mid+1;
            }
            else high = mid-1;
        }
        return (int)high;
    }
}

File: Difficulty: Easy/Square Root/square-root.cpp
class Solution {
  public:
    int floorSqrt(int n) {
        // code here
        int low = 0;
        int high = n;
        
        while(low<=high) {
            long long mid = low + (high - low) / 2;
            long long square = mid * mid;
            
            if(square <= n) low = mid + 1;
            else high = mid-1;
        }
        return high;
    }
};